.github/workflows/test.yml

```
@@ -0,0 +1,23 @@
+name: Unit Tests
+
+on: [push, pull_request]
+
+jobs:
+  tests:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            components: rustfmt, clippy
+      - run: |
+          cargo test
+      - uses: codecov/codecov-action@v3
+        with:
+          files: ./test_results/opensearch.lcov
+          flags: integration
```

`cargo test` on its own will not collect coverage data, you need to configure profiling and then convert that data into the final lcov.

```suggestion
name: Unit Tests

on: [push, pull_request]

jobs:
  tests:
    name: Test
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
    env:
      TEST_RESULTS_PATH: ${{ github.workspace }}/test_results
      LCOV_PATH: ${{ env.TEST_RESULTS_PATH }}/opensearch.lcov
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Install latest stable toolchain
        uses: actions-rs/toolchain@v1
        with:
            toolchain: stable
            components: rustfmt, clippy, llvm-tools-preview
      
      - name: Setup Tools
        run: |
          mkdir -p $TEST_RESULTS_PATH \
          && cargo install grcov

      - name: Run Tests
        run: cargo test
        env:
          RUSTFLAGS: -C instrument-coverage
          LLVM_PROFILE_FILE: ${{ env.TEST_RESULTS_PATH }}/opensearch-%m.profraw

      - name: Convert Coverage Data
        run: |
          grcov $TEST_RESULTS_PATH \
            -s . \
            --binary-path ./target/debug \
            -t lcov \
            --branch \
            --ignore-not-existing \
            --llvm \
            -o $LCOV_PATH

      - name: Upload Coverage Data
        uses: codecov/codecov-action@v3
        with:
          files: ${{ env.LCOV_PATH }}
          flags: unit
```

---

An alternative is to add a task like below to the `Makefile.toml` file:
```toml
[tasks.run-unittest]
private = true
command = "cargo"
args = ["test"]

[tasks.unittest]
category = "OpenSearch"
clear = true
description = "Runs workspace unit tests"
dependencies = ["set-coverage-vars", "create-test-results-dir", "run-unittest"]
run_task = "convert-coverage-data"
```

and then change workflow like so:
```suggestion
name: Unit Tests

on: [push, pull_request]

jobs:
  tests:
    name: Test
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
    steps:
      - name: Checkout
        uses: actions/checkout@v3

      - name: Install latest stable toolchain
        uses: actions-rs/toolchain@v1
        with:
            toolchain: stable
            components: rustfmt, clippy, llvm-tools-preview
      
      - name: Setup Tools
        run: cargo install grcov

      - name: Run Tests
        run: cargo make unittest

      - name: Upload Coverage Data
        uses: codecov/codecov-action@v3
        with:
          files: ./test_results/opensearch.lcov
          flags: unit
```